vb-fe/src/components/player/PlayerWrapper.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { FunctionComponent, useEffect } from "react";
+import { useDispatch } from "react-redux";
+import { usePlayerDevice, useErrorState, usePlaybackState } from "core/player/index";
+import { PlayerConstants } from "_store/player/PlayerTypes";
+import { SystemConstants } from "_store/system/SystemTypes";
+import { resetApp } from "util/Logout";
+
+export const PlayerWrapper: FunctionComponent = () => {
+  const device = usePlayerDevice();
+  const error = useErrorState();
+  const playbackState = usePlaybackState();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch({ type: SystemConstants.LOADING, payload: "Initializing Player..." });
+    if (device) {
+      dispatch({ type: PlayerConstants.SET_DEVICE_ID, payload: device.device_id });
+      dispatch({ type: SystemConstants.SUCCESS });
+    }
+  }, [device, dispatch]);
+
+  useEffect(() => {
+    if (error) {
+      switch (error.type) {
+        case "authentication_error":
+          resetApp();
+          dispatch({ type: SystemConstants.LOGIN, payload: false });
+          break;
+        default:
+          resetApp();
+          dispatch({ type: SystemConstants.LOGIN, payload: false });
+      }
+      dispatch({ type: SystemConstants.FAILURE });
+    }
+  }, [error, dispatch]);
+
+  useEffect(() => {
+    if (playbackState) {
+      console.log(playbackState);
+      dispatch({
+        type: PlayerConstants.UPDATE_TRACK,
+        payload: playbackState.track_window.current_track,
+      });
+      dispatch({
+        type: playbackState.paused ? PlayerConstants.PAUSE : PlayerConstants.PLAY,
+      });
+      dispatch({
+        type: PlayerConstants.UPDATE_POSITION,
+        payload: playbackState.position,
+      });
+      dispatch({ type: PlayerConstants.SET_SHUFFLE, payload: playbackState.shuffle });
```

Comment:
Intuitively it seems unusual to dispatch all of these on every `playbackState` update, is there a reason why?

Reply:
Cause you need to know whats happening with the player. This callback only takes place when something in the player changes. Since its an SDK that we're keeping track on our frontend we need to make sure it syncs up with what the player state is at. So yea, i'm guessing in teh future there will be more